test/e2e/util/mongotester/mongotester_test.go

```diff
@@ -0,0 +1,41 @@
+package mongotester
+
+import (
+	"crypto/tls"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+func TestTlsRemoval_RemovesCorrectConfig(t *testing.T) {
+	var opts []*options.ClientOptions
+
+	// configure TLS and hosts
+	opts = withTls(&tls.Config{
+		ServerName: "some-name",
+	}).ApplyOption(opts...)
+	opts = WithHosts([]string{"host1", "host2", "host3"}).ApplyOption(opts...)
+
+	removalOpt := WithoutTls()
+
+	// remove the tls value
+	opts = removalOpt.ApplyOption(opts...)
+
+	assert.Len(t, opts, 1, "tls removal should remove an element")
+	assert.NotNil(t, opts[0].Hosts, "tls removal should not effect other configs")
+	assert.Len(t, opts[0].Hosts, 3, "original configs should not be changed")
```

Comment:
[nit] just to be super-safe, can we check that `opts[0].Hosts` is exactly equal to the original one?